### PR TITLE
alsa-scarlett-gui: init at unstable-2022-08-11

### PR DESCRIPTION
**Pull request by SebTM**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/alsa-scarlett-gui/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    repo = pname;
+    rev = "65c0f6aa432501355803a823be1d3f8aafe907a8";
+    sha256 = "sha256-wzBOPTs8PTHzu5RpKwKhx552E7QnDx2Zn4OFaes8Q2I=";
+  };
+
+  makeFlags = [
+    "DESTDIR=\${out}"
+    "PREFIX=''"
+  ];
+  postUnpack = "sourceRoot=$sourceRoot/src";
```
Comment (muscaln) [inline pkgs/applications/audio/alsa-scarlett-gui/default.nix:25]:
```suggestion
  sourceRoot = "source/src";
```

File: pkgs/applications/audio/alsa-scarlett-gui/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  nativeBuildInputs = [
+    alsa-lib
+    gtk4
+    pkg-config
+    wrapGAppsHook4
+  ];
```
Comment (muscaln) [inline pkgs/applications/audio/alsa-scarlett-gui/default.nix:31]:
```suggestion
  nativeBuildInputs = [ pkg-config wrapGAppsHook4 ];
  buildInputs = [ gtk4 alsa-lib ];
```

**Review by muscaln — CHANGES_REQUESTED** [review]

**Comment by SebTM** [comment]
Thanks for the feedback, I've adjusted the PR :)

File: pkgs/applications/audio/alsa-scarlett-gui/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{ lib, alsa-lib, fetchFromGitHub, gtk4, pkg-config, stdenv, wrapGAppsHook4 }:
```
Comment (muscaln) [inline pkgs/applications/audio/alsa-scarlett-gui/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, pkg-config
, alsa-lib
, gtk4
, wrapGAppsHook4
}:
```

File: pkgs/applications/audio/alsa-scarlett-gui/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  makeFlags = [ "DESTDIR=\${out}" "PREFIX=''" ];
+  sourceRoot = "source/src";
+  nativeBuildInputs = [ pkg-config wrapGAppsHook4 ];
+  buildInputs = [ gtk4 alsa-lib ];
+
+  meta = with lib; {
+    description =
+      "alsa-scarlett-gui is a Gtk4 GUI for the ALSA controls presented by the Linux kernel Focusrite  Scarlett Gen 2/3 Mixer Driver";
```
Comment (muscaln) [inline pkgs/applications/audio/alsa-scarlett-gui/default.nix:21]:
```suggestion
    description = "GUI for alsa controls presented by Focusrite Scarlett Gen 2/3 Mixer Driver";
```

**Review by muscaln — CHANGES_REQUESTED** [review]

**Review by muscaln — APPROVED** [review]

**Pull request merged**